### PR TITLE
feat: LT-1121 - Super Users and Administer Role Permission

### PR DESCRIPTION
**Pull request by cstobart**

### CODE REVIEW

File: conf/constants.py [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
 class Permissions:
     MANAGE_FINAL_ADVICE = 'MANAGE_FINAL_ADVICE'
     MANAGE_TEAM_ADVICE = 'MANAGE_TEAM_ADVICE'
     REVIEW_GOODS = 'REVIEW_GOODS'
+    ADMINISTER_ROLES = 'ADMINISTER_ROLES'
+
+
+class Roles:
+    SUPER_USER_ROLE_ID = UUID('00000000-0000-0000-0000-000000000002')
+    DEFAULT_ROLE_ID = UUID('00000000-0000-0000-0000-000000000001')
```
Comment (jfaracik) [inline conf/constants.py]: Could these be inverted to be in UUID order?

File: goods/tests/tests_control_codes.py [hunk 2]
```diff
@@ -165,12 +166,18 @@ def test_controlled_good_empty_control_code(self):
         verified_good = Good.objects.get(pk=self.good_1.pk)
         self.assertEqual(verified_good.flags.count(), 1)
 
-    # User must have permission to create team advice
+    # User must have permission to review goods
+    @tag('only')
```
Comment (jfaracik) [inline goods/tests/tests_control_codes.py]: Remove tag and comment above function

File: gov_users/urls.py [hunk 3]
```diff
@@ -8,9 +8,9 @@
     path('', gov_user_views.GovUserList.as_view(), name='gov_users'),
     path('authenticate/', gov_user_views.AuthenticateGovUser.as_view(), name='authenticate'),
     path('<uuid:pk>/', gov_user_views.GovUserDetail.as_view(), name='gov_user'),
-    path('roles/', roles_views.Roles.as_view(), name='roles'),
+    path('roles/', roles_views.RolesViews.as_view(), name='roles'),
```
Comment (jfaracik) [inline gov_users/urls.py]: Could you rename `roles_views` to `roles`?

File: conf/constants.py (hunk 1 above)
Comment (cstobart) [reply conf/constants.py]: sure

Comment (cstobart) [reply conf/constants.py]: done

File: goods/tests/tests_control_codes.py (hunk 2 above)
Comment (cstobart) [reply goods/tests/tests_control_codes.py]: done

File: gov_users/urls.py (hunk 3 above)
Comment (cstobart) [reply gov_users/urls.py]: done

File: cases/tests/tests_final_decisions.py [hunk 4]
```diff
@@ -17,6 +17,8 @@ def setUp(self):
         self.url = reverse('cases:activity', kwargs={'pk': self.case.id})
 
     def test_cannot_make_final_decision_without_permission(self):
+        self.gov_user.role = self.default_role
```
Comment (markm5A8Z9F43913A) [inline cases/tests/tests_final_decisions.py]: Should this not be the rule rather than the exception?

Comment (cstobart) [reply cases/tests/tests_final_decisions.py]: testing everything with permissions

File: gov_users/tests/tests_roles_and_permissions.py [hunk 5]
```diff
@@ -44,6 +48,19 @@ def test_get_list_of_all_roles(self):
...
+        self.gov_user.save()
+        role = Role(name='some')
+        role.permissions.set([Permissions.MANAGE_FINAL_ADVICE])
+        role.save()
+
+        response = self.client.get(self.url, **self.gov_headers)
+        response_data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response_data['roles']), 3)
```
Comment (markm5A8Z9F43913A) [inline gov_users/tests/tests_roles_and_permissions.py]: Will it always be 3 or will it be previous value + 1?

File: gov_users/tests/tests_roles_and_permissions.py [hunk 6]
```diff
@@ -82,9 +101,33 @@ def test_edit_a_role(self):
...
     ])
     def test_role_name_must_be_unique(self, data):
+        self.gov_user.role = self.super_user_role
+        self.gov_user.save()
         Role(name='this is a name').save()
 
         response = self.client.post(self.url, data, **self.gov_headers)
 
-        self.assertEqual(Role.objects.all().count(), 2)
+        self.assertEqual(Role.objects.all().count(), 3)
```
Comment (markm5A8Z9F43913A) [inline gov_users/tests/tests_roles_and_permissions.py]: Will it always be 3 or will it be previous value + 1?

**Pull request merged**